cmd/mattermost/commands/team.go

```diff
@@ -339,3 +353,43 @@ func archiveTeamCmdF(command *cobra.Command, args []string) error {
 
 	return nil
 }
+
+func modifyTeamCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+	username, err := command.Flags().GetString("username")
+	if username == "" || err != nil {
+		return errors.New("Username is required.")
+	}
+
+	public, _ := command.Flags().GetBool("public")
+	private, _ := command.Flags().GetBool("private")
+
+	if public == private {
+		return errors.New("You must specify only one of --public or --private")
+	}
+
+	team := getTeamFromTeamArg(a, args[0])
+	if team == nil {
+		return errors.New("Unable to find team '" + args[0] + "'")
+	}
+
+	if !(team.Type == model.TEAM_OPEN || team.Type == model.TEAM_PRIVATE) {
+		return errors.New("You can only change the type of public/private teams.")
+	}
+
+	team.Type = model.TEAM_OPEN
+	if private {
+		team.Type = model.TEAM_PRIVATE
+	}
+
+	user := getUserFromUserArg(a, username)
```

Comment:
I think we need to move this up next to username validation and have the user validated as well. Exit early if user == nil.